tests/dependencies/test_dependency.py

```diff
@@ -59,16 +54,34 @@ def get_test_route(request: Request):
         return_value=jwks_fake_data,
     )
     mocked_jwt.start()
-    test_app.add_middleware(JWKSAuthMiddleware, jwks_validator=jwks_verifier)
-    yield test_app
+    yield JWKSAuth(jwks_validator=jwks_verifier)
     mocked_jwt.stop()
 
 
+@pytest.fixture()
+def app(jwks_auth: JWKSAuth) -> FastAPI:
+    test_app = FastAPI(dependencies=[Security(jwks_auth)])
+
+    @test_app.get("/test-endpoint", response_model=FakeToken)
+    def get_test_route(request: Request):
+        return request.state.payload
+
+    return test_app
+
+
 @pytest.fixture()
 def client(app: FastAPI) -> TestClient:
     return TestClient(app)
 
 
+def test_openapi_security_schema(app: FastAPI):
+    openapi = app.openapi()
+    assert openapi["components"]["securitySchemes"] == {
+        "JWKSAuth": {"scheme": "bearer", "type": "http"}
```

Comment:
I assume `JWKSAuth` is generated by FastAPI using the class name? Is there a way to make this configurable?

Reply:
yes, class name is the fallback but it's possible to customize it using the `scheme_name` parameter. I've updated the test case